sql/api/src/main/scala/org/apache/spark/sql/types/StructType.scala

```diff
@@ -377,6 +377,62 @@ case class StructType(fields: Array[StructField]) extends DataType with Seq[Stru
     findFieldInStruct(this, fieldNames, Nil)
   }
 
+  /**
+   * Returns a field in this struct and its child structs, case insensitively. This is slightly less
+   * performant than the case sensitive version.
+   *
+   * If includeCollections is true, this will return fields that are nested in maps and arrays.
+   *
+   * @param fieldNames The path to the field, in order from the root. For example, the column
+   *                   nested.a.b.c would be Seq("nested", "a", "b", "c").
+   */
+  private[spark] def findNestedFieldIgnoreCase(
```

Comment:
Looks like we can simply call the existing `findNestedField` and pass a case insensitive resolver

Reply:
> pass a case insensitive resolver

any example code ?